src/DS/rds.c

```diff
@@ -758,22 +823,45 @@ static int ds_rds_create_from_dom(xmlDocPtr* ret, xmlDocPtr sds_doc, xmlDocPtr x
 	return 0;
 }
 
-struct oscap_source *ds_rds_create_source(struct oscap_source *sds_source, struct oscap_source *xccdf_result_source, struct oscap_htable *oval_result_sources, struct oscap_htable *oval_result_mapping, struct oscap_htable *arf_report_mapping, const char *target_file)
+struct oscap_source *ds_rds_create_source(struct oscap_source *sds_source, struct oscap_source *tailoring_source, struct oscap_source *xccdf_result_source, struct oscap_htable *oval_result_sources, struct oscap_htable *oval_result_mapping, struct oscap_htable *arf_report_mapping, const char *target_file)
 {
 	xmlDoc *sds_doc = oscap_source_get_xmlDoc(sds_source);
 	if (sds_doc == NULL) {
 		return NULL;
 	}
+
 	xmlDoc *result_file_doc = oscap_source_get_xmlDoc(xccdf_result_source);
 	if (result_file_doc == NULL) {
 		return NULL;
 	}
 
+	xmlDoc *tailoring_doc = NULL;
+	char *tailoring_doc_timestamp = NULL;
+	char *tailoring_filepath = NULL;
+	if (tailoring_source) {
+		tailoring_doc = oscap_source_get_xmlDoc(tailoring_source);
+		if (tailoring_doc == NULL) {
+			return NULL;
+		}
+		tailoring_filepath = oscap_source_get_filepath(tailoring_source);
```

Comment:
We need to solve the situation when tailoring isn't a separate file, but it is a part of a source datastream. In this situation, the `tailoring_filepath` is unknown and the subsequent `stat()` returns error. As a result, this function returns NULL, and finally ARF isn't created at all.